pkg/logs/internal/tailers/file/tailer.go

```diff
@@ -359,6 +380,53 @@ func getFormattedTime() string {
 	return fmt.Sprintf("%s / %s (%d)", local, utc, milliseconds)
 }
 
+// NewMovingSum initializes a new MovingSum.
+func NewMovingSum(timeWindow int64, getTimeFunc func() int64) *MovingSum {
+	return &MovingSum{
+		DataPoints:   make([]*DataPoint, 0),
+		TimeWindow:   timeWindow,
+		timeProvider: getTimeFunc,
+		mu:           &sync.Mutex{},
+	}
+}
+
+// AddDataPoint adds a new data point to the moving sum.
+func (ms *MovingSum) AddDataPoint(value int64) {
+	ms.mu.Lock()
+	defer ms.mu.Unlock()
+
+	now := ms.timeProvider()
+	ms.DataPoints = append(ms.DataPoints, &DataPoint{
```

Comment:
We should probably bucket these into configurable bucket sizes (similar to `StatsTracker`) otherwise `DataPoints` can grow very large on a busy file. 

It shouldn't be a very large addition to your existing code :)